src/main/java/se/sundsvall/emailreader/service/EmailScheduler.java

```diff
@@ -57,23 +63,24 @@ public void checkForNewEmails() {
 			final var result = ewsIntegration
 				.pageThroughEntireInbox(credential.getUsername(), encryptionUtility.decrypt(credential.getPassword()), credential.getDomain(), emailAddress);
 
+			handleNewEmails(result, emailAddress, credential);
+		}));
+	}
 
-			result.forEach(email -> {
-
-				try {
-					emailRepository.save(EmailMapper.toEmailEntity(email, credential.getNamespace(), credential.getMunicipalityId(), credential.getMetadata()));
-				} catch (final Exception e) {
-					log.error("Failed to save email", e);
-					return;
-				}
-
-				try {
+	void handleNewEmails(final List<Email> emails, final String emailAddress, final CredentialsEntity credential){
+		for(var email : emails){
+			var newEmail = new EmailEntity();
+			try {
+				newEmail = emailRepository.save(EmailMapper.toEmailEntity(email, credential.getNamespace(), credential.getMunicipalityId(), credential.getMetadata()));
+				if(newEmail.getId() != null){
 					ewsIntegration.moveEmail(ItemId.getItemIdFromString(email.id()), emailAddress, credential.getDestinationFolder());
-				} catch (final Exception e) {
-					log.error("Failed to move email", e);
 				}
-			});
-		}));
+			} catch (final Exception e) {
+				log.error("Failed to save email", e);
+				emailRepository.deleteById(newEmail.getId());
```

Comment:
This could will throw exception if save action in the try is the thing throwing the original exception.